src/components/select-field/BaseSelectField.js

```diff
@@ -381,34 +399,47 @@ class BaseSelectField extends React.Component<Props, State> {
             listboxProps['aria-multiselectable'] = true;
         }
 
+        const dropdownPlacement = isRightAligned ? 'bottom-end' : 'bottom-start';
+
         return (
             // eslint-disable-next-line jsx-a11y/no-static-element-interactions
             <div
                 className={classNames(className, 'select-container')}
                 onBlur={this.handleBlur}
                 onKeyDown={this.handleKeyDown}
             >
-                <div className="select-field">
-                    {this.renderSelectButton()}
-                    <div
-                        className={classNames(OVERLAY_WRAPPER_CLASS, {
-                            'is-visible': isOpen,
-                        })}
-                    >
-                        <ul
-                            className={classNames('overlay', {
-                                [OVERLAY_SCROLLABLE_CLASS]: isScrollable,
-                            })}
-                            id={this.selectFieldID}
-                            role="listbox"
-                            // preventDefault on mousedown so blur doesn't happen before click
-                            onMouseDown={event => event.preventDefault()}
-                            {...listboxProps}
-                        >
-                            {this.renderSelectOptions()}
-                        </ul>
-                    </div>
-                </div>
+                <Manager>
+                    <Reference>
+                        {({ ref }) =>
+                            // Need to store the select button reference so we can calculate the button width
+                            // in order to set it as the min width of the dropdown list
+                            React.cloneElement(this.renderSelectButton(), { ref: this.setSelectButtonRef(ref) })
```

Comment:
Or can we just do this in the renderSelectButton()
```jsx
<SelectButton {...buttonProps} error={error} ref={this.selectButtonRef}>
    {buttonText}
</SelectButton>
```

Reply:
I'll try passing it into `renderSelectButton`, I'm not convinced that I can use `React.createRef` for selectButtonRef because I need to invoke Popper's refHandler to associate the popper with the button, see my reply here: https://github.com/box/box-ui-elements/pull/1715/files#r344312163